core/storage/statistics/gae_models.py

```diff
@@ -587,114 +590,219 @@ def get_versions(cls, exploration_id):
             ).fetch(feconf.DEFAULT_QUERY_LIMIT)]
 
 
-def process_submitted_answer(
-        exploration_id, unused_exploration_version, state_name,
-        rule_spec_string, answer):
-    """Adds an answer to the answer log for the rule it hits.
-
-    Args:
-        exploration_id: the exploration id
-        state_name: the state name
-        answer: an HTML string representation of the answer
-    """
-    # Add answer to StateRuleAnswerLogModel
-    answer_log = StateRuleAnswerLogModel.get_or_create(
-        exploration_id, state_name, rule_spec_string)
-    if answer in answer_log.answers:
-        answer_log.answers[answer] += 1
-    else:
-        answer_log.answers[answer] = 1
-
-    # This may fail due to answer_log.answers being larger than 1 MB in size.
-    try:
-        answer_log.put()
-    except Exception as e:
-        logging.error(e)
-
-
-def resolve_answers(
-        exploration_id, state_name, rule_str, answers):
-    """Resolves selected answers for the given rule.
-
-    Args:
-        exploration_id: the exploration id
-        state_name: the state name
-        rule_str: a string representation of the rule
-        answers: a list of HTML string representations of the resolved answers
-    """
-    assert isinstance(answers, list)
-    answer_log = StateRuleAnswerLogModel.get_or_create(
-        exploration_id, state_name, rule_str)
-
-    for answer in answers:
-        if answer not in answer_log.answers:
-            logging.error(
-                'Answer %s not found in answer log for rule %s of exploration '
-                '%s, state %s, handler %s' % (
-                    answer, rule_str, exploration_id, state_name,
-                    _OLD_SUBMIT_HANDLER_NAME))
-        else:
-            del answer_log.answers[answer]
-    answer_log.put()
-
-
 class StateAnswersModel(base_models.BaseModel):
-    """Store all answers of a state.
+    """Store all answers of a state. This model encapsulates a sharded storage
+    system for answers. Multiple entries in the model may contain answers for
+    the same state. The initial entry has a shard ID of 0 and contains
+    information about how many shards exist for this state. All other meta
+    information is duplicated across all shards, since they are immutable or are
+    local to that shard.
 
     The id/key of instances of this class has the form
-        [EXPLORATION_ID]:[EXPLORATION_VERSION]:[STATE_NAME].
+        [EXPLORATION_ID]:[EXPLORATION_VERSION]:[STATE_NAME]:[SHARD_ID].
     """
+    # This provides about 48k of padding for the other properties and entity
+    # storage overhead (since the max entity size is 1MB).
+    _MAX_ANSWER_LIST_BYTE_SIZE = 100000
```

Comment:
You are missing nothing, I am apparently missing a zero. :) This might have been changed for tests, I'm not sure. The comment was also wrong and I updated it.